File: LeetCode/Python Solutions/Check If Array Pairs Are Divisible by k/check.py
class Solution:
    def canArrange(self, arr: List[int], k: int) -> bool:
        remainder_count = [0] * k
        
        # Count the frequency of each remainder
        for num in arr:
            remainder = num % k
            remainder_count[remainder] += 1
        
        # Check if the array can be arranged into pairs
        for i in range(k // 2 + 1):
            if i == 0:
                # Remainder 0 should have even count
                if remainder_count[i] % 2 != 0:
                    return False
            elif i == k - i:
                # For k/2 (when k is even), count should be even
                if remainder_count[i] % 2 != 0:
                    return False
            else:
                # For other remainders, count should match with its complement
                if remainder_count[i] != remainder_count[k - i]:
                    return False
        
        return True

File: LeetCode/Java Solutions/Check If a Word Occurs As a Prefix of Any Word in a Sentence/check.java
class Solution {
    public boolean canArrange(int[] arr, int k) {
        int[] remainderCount = new int[k];
        
        for (int num : arr) {
            int remainder = ((num % k) + k) % k;  // Handle negative numbers
            remainderCount[remainder]++;
        }
        
        if (remainderCount[0] % 2 != 0) {
            return false;
        }
        
        for (int r = 1; r <= k / 2; r++) {
            if (r * 2 == k) {
                if (remainderCount[r] % 2 != 0) {
                    return false;
                }
            } else if (remainderCount[r] != remainderCount[k - r]) {
                return false;
            }
        }
        
        return true;
    }
}